### PR TITLE
648 and 647: Populate serverVersion in product catalogue when saving and make two fields optional in the API

### PR DESCRIPTION
**Pull request by rehammuzzamil**
https://github.com/OpenSRP/opensrp-server-web/issues/648
https://github.com/OpenSRP/opensrp-server-web/issues/647

### CODE REVIEW

File: src/main/java/org/opensrp/service/ProductCatalogueService.java [hunk 1]
```diff
@@ -33,11 +33,13 @@ public ProductCatalogue getProductCatalogue(Long uniqueId) {
 
 	public void add(ProductCatalogue productCatalogue) {
 		validateFields(productCatalogue);
+		productCatalogue.setServerVersion(productCatalogueRepository.getNextServerVersion());
 		productCatalogueRepository.add(productCatalogue);
 	}
 
 	public void update(ProductCatalogue productCatalogue) {
 		validateFields(productCatalogue);
+		productCatalogue.setServerVersion(productCatalogueRepository.getNextServerVersion());
```
Comment (githengi) [inline src/main/java/org/opensrp/service/ProductCatalogueService.java:42]: You need to define Sequence in productCatalogueRepository for this to work

Comment (rehammuzzamil) [reply src/main/java/org/opensrp/service/ProductCatalogueService.java:42]: Done

File: pom.xml [hunk 2]
```diff
@@ -24,7 +24,7 @@
 		<opensrp.common.version>2.0.3-SNAPSHOT</opensrp.common.version>
 		<opensrp.api.version>1.0.4-SNAPSHOT</opensrp.api.version>
 		<opensrp.interface.version>2.0.1-SNAPSHOT</opensrp.interface.version>
-		<opensrp.plan.evaluator.version>1.0.2-SNAPSHOT</opensrp.plan.evaluator.version>
+		<opensrp.plan.evaluator.version>1.0.3-SNAPSHOT</opensrp.plan.evaluator.version>
```
Comment (githengi) [inline pom.xml:27]: Update to the latest version 1.0.4

**Review by githengi — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (rehammuzzamil) [reply pom.xml]: Done

**Pull request merged**